diff --git a/SConstruct b/SConstruct
index 31a8856..18d73bf 100644
--- a/SConstruct
+++ b/SConstruct
@@ -301,14 +301,18 @@ def config_base(env):
 
     try:
         BOOST_ROOT = os.path.normpath(os.environ['BOOST_ROOT'])
+        env.Append(CPPPATH=[
+            os.path.join(BOOST_ROOT, 'include'),
+            ])
         env.Append(LIBPATH=[
-            os.path.join(BOOST_ROOT, 'stage', 'lib'),
+            os.path.join(BOOST_ROOT, 'lib')
             ])
         env['BOOST_ROOT'] = BOOST_ROOT
+        env.Prepend(LINKFLAGS=['-Wl,-rpath=%s'%os.path.join(BOOST_ROOT, 'lib')])
     except KeyError:
         pass
 
-    if Beast.system.windows:
+    if not Beast.system.osx:
         try:
             OPENSSL_ROOT = os.path.normpath(os.environ['OPENSSL_ROOT'])
             env.Append(CPPPATH=[
@@ -317,6 +321,7 @@ def config_base(env):
             env.Append(LIBPATH=[
                 os.path.join(OPENSSL_ROOT, 'lib'),
                 ])
+            env.Prepend(LINKFLAGS=['-Wl,-rpath=%s'%os.path.join(OPENSSL_ROOT, 'lib')])
         except KeyError:
             pass
     elif Beast.system.osx:
